modules/azure/compute.go

```diff
@@ -8,97 +8,451 @@ import (
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
 	if err != nil {
 		return "", err
 	}
 
-	// Get the details of the target virtual machine
-	vm, err := vmClient.Get(context.Background(), resGroupName, vmName, compute.InstanceView)
+	// Virtual Machine has no associated Availability Set
+	if vm.AvailabilitySet == nil {
+		return "", nil
+	}
+
+	return GetNameFromResourceID(*vm.AvailabilitySet.ID), nil
+}
+
+// VMImage represents the storage image for the specified Azure Virtual Machine.
+type VMImage struct {
+	Publisher string
+	Offer     string
+	SKU       string
+	Version   string
+}
+
+// GetVirtualMachineImage gets the Image of the specified Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineImage(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) VMImage {
+	vmImage, err := GetVirtualMachineImageE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return vmImage
+}
+
+// GetVirtualMachineImageE gets the Image  of the specified Azure Virtual Machine.
+func GetVirtualMachineImageE(vmName string, resGroupName string, subscriptionID string) (VMImage, error) {
+	var vmImage VMImage
+
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		return vmImage, err
+	}
+
+	if vm.StorageProfile == nil {
+		return vmImage, NewNotFoundError("Image Reference", "Any", vmName)
+	}
+
+	// Populate VM Image; values always present, no nil checks needed
+	vmImage.Publisher = *vm.StorageProfile.ImageReference.Publisher
+	vmImage.Offer = *vm.StorageProfile.ImageReference.Offer
+	vmImage.SKU = *vm.StorageProfile.ImageReference.Sku
+	vmImage.Version = *vm.StorageProfile.ImageReference.Version
+
+	return vmImage, nil
+}
+
+// GetVirtualMachineAdminUser gets the Admin Username of the specified Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineAdminUser(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) string {
+	adminUser, err := GetVirtualMachineAdminUserE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return adminUser
+}
+
+// GetVirtualMachineAdminUserE gets the Admin Username of the specified Azure Virtual Machine.
+func GetVirtualMachineAdminUserE(vmName string, resGroupName string, subscriptionID string) (string, error) {
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		return "", err
+	}
+
+	return string(*vm.OsProfile.AdminUsername), nil
+}
+
+// GetVirtualMachineSize gets the Size Type of the specified Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineSize(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) compute.VirtualMachineSizeTypes {
```

Comment:
Is this rename necessary? The reason I ask is that this would be a backwards incompatible change, and I am not sure the name change is worth the backwards incompatibility.

Reply:
Not necessary, reverted.